woocommerce/woocommerce-android
# Blaze: loading and error screens

Closes: #10808 


### Description
This PR address three parts:
1. Logic of campaign creation in the ViewModel.
2. Loading and error UI.
3. Handles retrying when an image upload fails (p1707821290862699/1704188207.780169-slack-C03L1NF1EA3)

### Testing instructions
1. Start campaign creation.
2. Choose a local image.
3. Proceed to creating the campaign.
4. Confirm that the creation succeeds (it uses a mocked API call)
5. Check in wp-admin and confirm that the image was uploaded.

### Images/gif
| Loading | Error |
| --- | --- |
| ![Screenshot_20240215_121047](https://github.com/woocommerce/woocommerce-android/assets/1657201/8ead550c-ee75-41bc-abab-3b401ed2e104) | ![Screenshot_20240215_121636](https://github.com/woocommerce/woocommerce-android/assets/1657201/be3dd6b8-a671-4a7c-b446-68314da7fff8) |
